engine/notifications.js

```diff
@@ -0,0 +1,273 @@
+    const numberOfFeatures = features.length;
+    const message = 'Registered to';
+    if (features.indexOf('all') >= 0) {
+      payload.body = `${message} all features`;
+    } else if (features.indexOf('new') >= 0) {
+      payload.body = `${message} new features only`;
+    } else if (numberOfFeatures === 1) {
+      payload.body = `${message} one feature`;
+    } else {
+      payload.body = `${message} ${numberOfFeatures} features`;
+    }
+  }
+  return sendWebPush(device, payload);
+}
+
+function unregisterDevice(deviceId) {
+  return redis.del(`device-${deviceId}`)
+  .then(() => redis.smembers(`${deviceId}-notifications`))
+  .then(deviceFeatures => Promise.all(
+        deviceFeatures.map(slug => redis.srem(`${slug}-notifications`, deviceId))
+  ))
+  .then(() => redis.del(`${deviceId}-notifications`));
+}
+
+// unregisters from receiving notifications
+// required:
+// * deviceId
+// optional
+// * features
+// if no features provided unregister from all features and delete
+// endpoint entry
+function unregister(deviceId, features, doNotConfirm) {
+  let endpoint;
+  let key;
+  let authSecret;
+  return checkDeviceId(deviceId)
+  .then(() => redis.hgetall(`device-${deviceId}`))
+  .then(device => {
+    endpoint = device.endpoint;
+    key = device.key;
+    authSecret = device.authSecret;
+  })
+  .then(() => {
+    if (!features) {
+      return unregisterDevice(deviceId);
+    }
+    return redis.sismember(`${deviceId}-notifications`, 'all')
+    .then(registeredToAll => {
+      if (!registeredToAll) {
+        return Promise.all(features.map(slug => [
+          redis.srem(`${slug}-notifications`, deviceId),
+          redis.srem(`${deviceId}-notifications`, slug)])
+        );
+      }
+      // user decided to unregister from `all`
+      if (features.indexOf('all') >= 0) {
+        return redis.srem('all-notifications', deviceId)
+        .then(() => redis.srem(`${deviceId}-notifications`, 'all'));
+      }
+      // unregister from `all` and register to negative of `features`
+      return redis.srem('all-notifications', deviceId)
+      .then(() => redis.srem(`${deviceId}-notifications`, 'all'))
+      // register to everything except of features
+      .then(() => redis.get('status'))
+      .then(status => {
+        let allFeatures = ['new'];
+        allFeatures = allFeatures.concat(Object.keys(JSON.parse(status)));
+        return Promise.all(allFeatures.map(slug => {
+          if (features.indexOf(slug) >= 0) {
+            return Promise.resolve();
+          }
+          return redis.sadd(`${deviceId}-notifications`, slug)
+          .then(() => redis.sadd(`${slug}-notifications`, deviceId));
+        }));
+      });
+    });
+  })
+  .then(() => getRegisteredFeatures(deviceId))
+  .catch(err => {
+    if (err.message !== 'Not Found') {
+      throw err;
+    }
+    return [];
+  })
+  .then(regFeatures => {
+    if (!doNotConfirm && endpoint) {
+      return sendConfirmation({
```

Comment:
Nit: you could use the device that you got earlier, at https://github.com/mozilla/platform-status/pull/317/files#diff-2057481f5e0a7e8f1a1972ef82e339b0R89